sdk/go/common/apitype/service.go

```diff
@@ -56,3 +54,41 @@ type APICapabilityConfig struct {
 type CapabilitiesResponse struct {
 	Capabilities []APICapabilityConfig `json:"capabilities"`
 }
+
+// Represents feature-detected capabilities of the service the backend is connected to.
+type Capabilities struct {
+	// If non-nil, indicates that delta checkpoint updates are supported.
+	DeltaCheckpointUpdates *DeltaCheckpointUploadsConfigV2
+
+	// Indicates whether the service supports bulk encryption.
+	BulkEncryption bool
+}
+
+func (r CapabilitiesResponse) Parse() (Capabilities, error) {
```

Comment:
I think if this is going to live here, then we need `service_test.go` with some unit tests for it (we've added test coverage above, but I think we could exercise `Parse` independently here). Also we should add a doc comment for this, if only to help clarify `CapabilitiesResponse` vs `Capabilities` etc.